lib/menu.rb

```diff
@@ -0,0 +1,51 @@
+require_relative 'dish'
+require_relative 'formatter'
+
+class Menu
+  include Formatter
+
+  def initialize(dish_klass = Dish, dishes_str = nil)
+    @dish_klass = dish_klass
+    @dishes = Array.new
+    filename = "menu.txt"
+    dishes_str = read_file(filename) if dishes_str.nil?
+    parse_dishes(dishes_str)
+  end
+
+  def formatted_dishes
+    list = ""
+    @dishes.each_with_index do |dish, index|
+      num = (index).to_s
+      name = dish.name
+      price = pound_str(dish.price)
+      list << three_col_str(num, name, price)
+    end
+    list
+  end
+
+  def dishes
+    @dishes.dup
+  end
+
+  private
+
+  def read_file(filename)
+    if File.exist?(filename) && File.file?(filename)
```

Comment:
Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)